contracts/bond/CollateralWhitelist.sol

```diff
@@ -3,15 +3,25 @@ pragma solidity ^0.8.0;
 
 import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/IERC20MetadataUpgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/structs/EnumerableMapUpgradeable.sol";
 
 /**
  * @title Whitelist for collateral tokens.
  *
  * @notice Encapsulation of a ERC20 collateral tokens whitelist, indexed by their symbol.
  */
 abstract contract CollateralWhitelist is Initializable {
+    using EnumerableMapUpgradeable for EnumerableMapUpgradeable.UintToAddressMap;
+
     // Token symbols to ERC20 Token contract addresses
-    mapping(string => address) private _whitelist;
+    EnumerableMapUpgradeable.UintToAddressMap private _whitelist;
```

Comment:
Yeah, so looks like there's a reason string (unbounded array of bytes) are used for keys
https://docs.soliditylang.org/en/v0.8.5/internals/layout_in_storage.html#mappings-and-dynamic-arrays

Maybe consider using a hash of them and storing a set of string seperately?
https://ethereum.stackexchange.com/questions/101975/is-it-cheaper-to-store-a-string-or-convert-it-to-bytes32-with-abi-encodepacked

...as the uint is 256 bits in size, so if the string was every larger, the excess would be chopped off 🤔

Reply:
> Another potential issue is that symbol is OPTIONAL on ERC20, but likely youve already considered this. I do like a human readable key..

That one's easy, as it's already been solved, with [ERC20MetaData](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/interfaces/IERC20Metadata.sol) / [UpgradableVersion](https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable/blob/master/contracts/interfaces/IERC20MetadataUpgradeable.sol)

Using interfaces doesn't solve the issue that the EVM doesn't enforce contracts adhere to interfaces (i.e. when you receive an address, that the contract actually implements the function of the interface), but that's the EVM 🤷

Reply:
>Maybe the cleanest way to do this is to use bytes32 as the key everywhere and convert to string client side with ethers?

Yeah, a popular approach with strings is inputting bytes32 generated from a keeack hash of the string, with padding used ....but as it's hashing, that's only a one way street 😞 

I guess, this part is the exploratory aspect of this task 😁 

Alternatively we can keep the function sigs as strings, then internally use the hashing to generate bytes32, while also keeping a mapping of bytes32->string ....would that be workable?